src/ast/ast_enrich.c

```diff
@@ -1,405 +1,25 @@
-			// Recursively continue mapping.
-			_collect_projected_identifier(child, identifier_map);
-		}
-	}
-}
-
-static void _annotate_with_clause_projected_named_path(AST *ast,
-													   const cypher_astnode_t *with_clause, uint scope_start, uint scope_end) {
-	// Iterate over the projections and find their identifiers.
-	rax *identifier_map = raxNew();
-	uint with_projection_count = cypher_ast_with_nprojections(with_clause);
-	for(uint projection_iter = 0; projection_iter < with_projection_count; projection_iter++) {
-		const cypher_astnode_t *projection = cypher_ast_with_get_projection(with_clause, projection_iter);
-		_collect_projected_identifier(projection, identifier_map);
-	}
-	_annotate_relevant_projected_named_path_identifier(ast, identifier_map, scope_start, scope_end);
-	raxFreeWithCallback(identifier_map, array_free);
-}
-
-static void _annotate_return_clause_projected_named_path(AST *ast,
-														 const cypher_astnode_t *return_clause, uint scope_start, uint scope_end) {
-	rax *identifier_map = raxNew();
-	uint return_projection_count = cypher_ast_return_nprojections(return_clause);
-	for(uint projection_iter = 0; projection_iter < return_projection_count; projection_iter++) {
-		const cypher_astnode_t *projection = cypher_ast_return_get_projection(return_clause,
-																			  projection_iter);
-		_collect_projected_identifier(projection, identifier_map);
-	}
-	_annotate_relevant_projected_named_path_identifier(ast, identifier_map, scope_start, scope_end);
-	raxFreeWithCallback(identifier_map, array_free);
-}
-
-static void _annotate_projected_named_path(AST *ast) {
-	uint *with_clause_indices = AST_GetClauseIndices(ast, CYPHER_AST_WITH);
-	uint with_clause_count = array_len(with_clause_indices);
-	uint scope_start = 0;
-	uint scope_end;
-	// Handle with clauses.
-	for(uint i = 0; i < with_clause_count; i ++) {
-		scope_end = with_clause_indices[i];
-		const cypher_astnode_t *with_clause = cypher_ast_query_get_clause(ast->root, scope_end);
-		_annotate_with_clause_projected_named_path(ast, with_clause, scope_start, scope_end);
-		scope_start = scope_end;
-	}
-	array_free(with_clause_indices);
-	uint clause_count = cypher_ast_query_nclauses(ast->root);
-	const cypher_astnode_t *last_clause = cypher_ast_query_get_clause(ast->root, clause_count - 1);
-	cypher_astnode_type_t last_clause_type = cypher_astnode_type(last_clause);
-	if(last_clause_type == CYPHER_AST_RETURN) {
-		_annotate_return_clause_projected_named_path(ast, last_clause, scope_start, clause_count);
-	}
-}
+#include "enrichment/annotate_entities.h"
+#include "enrichment/annotate_params.h"
+#include "enrichment/annotate_project_all.h"
+#include "enrichment/annotate_projected_named_paths.h"
 
 //------------------------------------------------------------------------------
 //  Main AST enrichment
 //------------------------------------------------------------------------------
+/* Directives like CREATE INDEX are not queries. */
```

Comment:
I think this comment should be above the return at line 18.

Reply:
moved